todo/src/components/TodoForm.js

```diff
@@ -1,19 +1,46 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { addTodo } from '../actions';
 
-const TodoForm = props => {
-  return (
-    <form onSubmit={props.addTodo}>
-      <input
-      type="text"
-      onChange={props.handleChanges}
-      name="task"
-      value={props.task}
-      placeholder="...Todo"
-      required
-      />
-      <button type="submit">Add Todo</button>
-    </form>
-  )
+class TodoForm extends Component {
+  state = {
+    task: ''
+  }
+
+  addTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.task);
+    this.setState({ task: '' })
+  }
+
+  handleChanges = e => {
+    console.log(this.state)
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.addTodo}>
+        <input
+        type="text"
+        onChange={this.handleChanges}
+        name="task"
+        value={this.state.task}
+        placeholder="...Todo"
+        required
+        />
+        <button type="submit">Add Todo</button>
+      </form>
+    )
+  }
 }
 
-export default TodoForm;
+const mapStateToProps = (state) => {
+    return {
+        ...state
+    };
+};
```

Comment:
be careful you don't use a MSTP unnecessarily -- you don't need anything from the state tree in this component, you only need the actioncreator `addTodo`. you can pass `null` in place of the MSTP in connect here as well